controller_manager:
  ros__parameters:
    update_rate: 100 # Hz


    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster


    joint_trajectory_controller:
      type: "joint_trajectory_controller/JointTrajectoryController"

joint_trajectory_controller:
  ros__parameters:

    joints:
      - joint1
      - joint2
      - joint3
      - joint4
      - joint5
      - joint6
      - gripper_joint_left_1
      - gripper_joint_right_1

    # gains:
    #   gripper_joint_left_1:
    #     p_gain: 100.0
    #     i_gain: 0.1
    #     d_gain: 1.0
    #     i_clamp: 1
    #   gripper_joint_right_1:
    #     p_gain: 100.0
    #     i_gain: 0.1
    #     d_gain: 1.0
    #     i_clamp: 1

    write_op_modes:
      - joint1
      - joint2
      - joint3
      - joint4
      - joint5
      - joint6
      - gripper_joint_left_1
      - gripper_joint_right_1
    
    interface_name: position # position_effort #effort #position # velocity

    command_interfaces:
      - position
      #- velocity
    
    state_interfaces:
      - position
      - velocity
      
    state_publish_rate: 50.0 # Defaults to 50
    action_monitor_rate: 20.0 # Defaults to 20

    allow_partial_joints_goal: True
    hardware_state_has_offset: true
    deduce_states_from_derivatives: true
    
    constraints:
      stopped_velocity_tolerance: 0.01 # Defaults to 0.01
      goal_time: 0.0 # Defaults to 0.0 (start immediately)



